### ISSUE TITLE
Disabled options should be highlightable for accessibility

### ISSUE THREAD

**Issue by zkuzmic** [JedWatson/react-select#3354]
When navigating with a keyboard, disabled options should still be highlightable (but not selectable obviously) so that they can be read out by a screen reader. Without the ability to highlight the option, a screen reader user won't have any idea the disabled option exists.

An example of a Select with a disabled option:
https://codesandbox.io/s/232x2yzjkn

**Comment by sarahbethfederman** [comment]
@JedWatson was discussing this issue with @gwyneplaine and we agree that disabled options should be focusable as per @zkuzmic's comment.  This reflects the built-in <select> behavior. Generally disabled options are focusable (and announced they are "dimmed") but when attempting to select them, the <select> closes w/ the previously selected value still selected. 

It sounds like we will essentially be reversing <https://github.com/JedWatson/react-select/pull/1008>(this) change + blocking the selection of a disabled option. I will get started on this issue if we're all on board with this.
